### ISSUE TITLE
Allow Hydragnn models to be build with embedding layers for node and edge features

### ISSUE THREAD

**Issue by allaffa** [ORNL/HydraGNN#33]
class Net(torch.nn.Module):
        def __init__(self):
            super(Net, self).__init__()
    
            self.node_emb = Embedding(21, 21)
            self.edge_emb = Embedding(4, 4)
    
            aggregators = ['mean', 'min', 'max', 'std']
            scalers = ['identity', 'amplification', 'attenuation']
    
            self.convs = ModuleList()
            self.batch_norms = ModuleList()
            for _ in range(2):
                conv = PNAConv(in_channels=21, out_channels=21,
                               aggregators=aggregators, scalers=scalers, deg=deg,
                               edge_dim=4, towers=1, pre_layers=1, post_layers=1,
                               divide_input=False)
                self.convs.append(conv)
                self.batch_norms.append(BatchNorm(21))
    
            self.mlp = Sequential(Linear(21, 21), ReLU(), Linear(21, 21), ReLU(),
                                  Linear(21, 1))
